dev-utils/run-script.js

```diff
@@ -51,10 +63,71 @@ function launchSauceConnect() {
   console.log('set MODE=saucelabs to launch sauce connect')
 }
 
+function runE2eTests(configPath) {
+  const webDriverConfig = join(PROJECT_DIR, configPath)
+  runE2eTestsUtils(webDriverConfig, false)
+}
+
+function buildE2eBundles(basePath) {
+  buildE2eBundlesUtils(join(PROJECT_DIR, basePath), err => {
+    err && process.exit(2)
+  })
+}
+
+function runSauceTests(serve = 'true', path = './', ...scripts) {
+  /**
+   * `console.logs` from the tests will be truncated when the process exits
+   * To avoid truncation, we flush the data from stdout before exiting the process
+   */
+  if (process.stdout.isTTY && process.stdout._handle) {
+    process.stdout._handle.setBlocking(true)
+  }
+
+  let servers = []
```

Comment:
since you are already refactoring,  I would completely remove this `serve` flag and just start server by default.

Reply:
The serve flag is useful for development. e.g. if you want to only run the tests but don't want each time to take bring up a new server which takes time.

Reply:
I like to avoid too many moving parts in the code, We could just start the servers before and move the logical parts out.